parts/k8s/kubernetesmastergenerateproxycertscript.sh

```diff
@@ -47,9 +47,10 @@ write_certs_to_disk_with_retry() {
 # block until all etcd is ready
 retrycmd_if_failure 100 5 10 etcdctl cluster-health
 # Make etcd keys, adding a leading whitespace because etcd won't accept a val that begins with a '-' (hyphen)!
-if etcdctl mk $ETCD_REQUESTHEADER_CLIENT_CA " $(cat ${PROXY_CRT})"; then
-    etcdctl mk $ETCD_PROXY_KEY " $(cat ${PROXY_CLIENT_KEY})"
-    etcdctl mk $ETCD_PROXY_CERT " $(cat ${PROXY_CLIENT_CRT})"
+# etcdctl will output the data it's given, stdout is redirected to dev null to avoid capturing sensitive data in logs
+if etcdctl mk $ETCD_REQUESTHEADER_CLIENT_CA " $(cat ${PROXY_CRT})" > /dev/null 2>&1; then
```

Comment:
instead of redirecting the output, doing `set +x` then `set -x` after the sensitive part would suffice. See https://github.com/Azure/acs-engine/blob/master/parts/k8s/kubernetescustomscript.sh#L10 for an example.

Reply:
@CecileRobertMichon 

I borrowed from your example and tried this in an etcd container, however it still outputs to stdout. Are we sure this works as intended?

```sh
/ # cat test.sh
set +x
etcdctl mk /test "SENSITIVE DATA"
set -x
etcdctl mk /test2 "not sensitive data"
/ # sh test.sh
SENSITIVE DATA
+ etcdctl mk /test2 'not sensitive data'
not sensitive data
```

See how it still outputs "SENSITIVE DATA" to stdout?

Reply:
Oh you're right my bad. Thanks for the PR @ryanhartje, looks good!